lib/internal/url.js

```diff
@@ -1500,6 +1574,26 @@ function getPathFromURLPosix(url) {
   return decodeURIComponent(pathname);
 }
 
+function getPathBufferFromURLPosix(url) {
```

Comment:
Can we move this to C++? I'm afraid if we don't do it right now, no one will for the next 5 years.